sdk/python/kfp/_client.py

```diff
@@ -214,18 +220,38 @@ def _get_url_prefix(self):
     # In-cluster pod. We could use relative URL.
     return '/pipeline'
 
-  def create_experiment(self, name, description=None, namespace=''):
+  def set_user_namespace(self, namespace):
```

Comment:
This method seems to be a little confusing.
It's an instance method, but it changes a global setting.

Reply:
I think technically it's an instance setting as well, but synced into a global file. 
I can't think of any better way to solve this. But I guess I now understand your question about environment variable vs file storage.

Reply:
Changed context_setting to an instance variable.